### PR TITLE
added nat rules and ILB samples

### PR DESCRIPTION
**Pull request by ypitsch**

### CODE REVIEW

File: 201-2-vms-internal-load-balancer/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,321 @@
+﻿{
+    "$schema": "http://schema.management.azure.com/schemas/2014-04-01-preview/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "subscriptionId": {
+            "type": "string",
+            "metadata": {
+                "Description": "This is your subscriptionId"
```
Comment (johngossman) [inline 201-2-vms-internal-load-balancer/azuredeploy.json]: "description" not "Description"

File: 201-2-vms-internal-load-balancer/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+            }
+        },
+        "availabilitySetName": {
+            "type": "string",
+            "metadata": {
+                "Description": "This is the name of the Availability Set"
```
Comment (johngossman) [inline 201-2-vms-internal-load-balancer/azuredeploy.json]: For a 201 starter, we don't need to have this as a parameter. Just hardcode it somewhere.

File: 201-2-vms-internal-load-balancer/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+            "allowedValues": [
+                "Standard_LRS",
+                "Standard_GRS"
+            ],
+            "metadata": {
+                "Description": "This is the name of the storage account type (local or geo)"
+            }
+
+        },
+        "virtualNetworkName": {
```
Comment (johngossman) [inline 201-2-vms-internal-load-balancer/azuredeploy.json]: None of these names need to be parameters.

File: 201-2-vms-internal-load-balancer/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+                        },
+                        "privateIPAddress": "10.0.2.6",
+                        "privateIPAllocationMethod": "Static"
+                    },
+                    "Name": "ilbfrontend"
+                }],
+                "BackendAddressPools": [{
+                    "Name": "lbbackendpool"
+                }],
+                "LoadBalancingRules": [{
```
Comment (johngossman) [inline 201-2-vms-internal-load-balancer/azuredeploy.json]: camelCase not PascalCase

File: 201-2-vms-loadbalancer-natrules/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+            "type": "Microsoft.Network/loadBalancers",
+            "location": "[parameters('region')]",
+            "dependsOn": [
+                "nicLoop",
+                "[concat('Microsoft.Network/publicIPAddresses/', parameters('publicIPAddressName'))]"
+            ],
+            "properties": {
+                "frontendIPConfigurations": [
+                    {
+                        "name": "LBFE",
```
Comment (johngossman) [inline 201-2-vms-loadbalancer-natrules/azuredeploy.json]: How about FrontEnd? No abbreviations.

File: 201-2-vms-loadbalancer-natrules/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+        "publicIPAddressType": "Dynamic",
+
+
+
+        "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',parameters('vnetName'))]",
+        "subnetRef": "[concat(variables('vnetID'),'/subnets/',variables ('subnetName'))]",
+        "publicIPAddressID": "[resourceId('Microsoft.Network/publicIPAddresses',parameters('publicIPAddressName'))]",
+        "lbID": "[resourceId('Microsoft.Network/loadBalancers',parameters('lbName'))]",
+        "numberOfInstances": 2,
+        "nicId1": "[resourceId('Microsoft.Network/networkInterfaces',concat(parameters('nicNamePrefix'), 0))]",
```
Comment (johngossman) [inline 201-2-vms-loadbalancer-natrules/azuredeploy.json]: Doesn't look like this is required now that you are using loops

**Comment by singhkay** [comment]
Closing PR. Will submit a new one

**Pull request closed without merging**